tls/s2n_kex.c

```diff
@@ -156,6 +151,42 @@ static S2N_RESULT s2n_check_hybrid_ecdhe_kem(const struct s2n_cipher_suite *ciph
     return S2N_RESULT_OK;
 }
 
+static S2N_RESULT s2n_kex_configure_noop(const struct s2n_cipher_suite *cipher_suite,
+        struct s2n_connection *conn)
+{
+    return S2N_RESULT_OK;
+}
+
+static S2N_RESULT s2n_kex_unsupported(const struct s2n_cipher_suite *cipher_suite,
+        struct s2n_connection *conn, bool *is_supported)
+{
+    RESULT_ENSURE_REF(is_supported);
+    *is_supported = false;
```

Comment:
Maybe it would be useful to add a comment explaining why this is always false? My understanding is that the tls 1.3 kex is going through a slightly different codepath, but at first glance it looks a little funky for this to always be false.

Reply:
I choose "always false" pretty arbitrarily, since you're right that TLS1.3 uses a different codepath so this was never being called.

But it could be called, and that removes move kex branching! I've switched it to actually check for TLS1.3. Alternatively it could always be true since we check for TLS1.3 at the cipher suite layer, but this seemed more correct.